### PR TITLE
[HEATMAP] Refactored: Data and Data preprocessing

### PR DESCRIPTION
**Pull request by uman95**
**_WHAT DOES THIS PR DO?_**

- Uses site data (as supposed to device data)
- Proper data preprocessing
- Remove spatial-temporal modelling for country-level AirQlouds

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [Refined spatial models for one international AirQloud](https://airqoteam.atlassian.net/browse/DP-205)
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

- create a virtual environment
- install all requirements, src/gpmodel/requirements.txt
- get the .env and test using development environment (**note that: for the event API we now use .../api/v2/..**)
- run the src/gpmodel/main.py

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

The [**heatmap** API](https://staging-platform.airqo.net/api/v1/predict/heatmap)

### CODE REVIEW

File: src/gp-model/data/data.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    site_ids = [site['_id'] for site in sites_details]
+
+    return site_ids
+
+
+def get_site_data(site_id, tenant='airqo'):
+    start_time = (datetime.utcnow()-timedelta(days=7)
+                  ).strftime('%Y-%m-%dT%H:%M:%SZ'),
+    end_time = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+    event_url = 'https://platform.airqo.net/api/v2/devices/events'
```
Comment (NoahNsimbe) [inline src/gp-model/data/data.py:35]: Hi @uman95, we can use an environment variable instead of hard coding `https://platform.airqo.net`

Comment (uman95) [reply src/gp-model/data/data.py]: Thanks @NoahNsimbe it was an oversight, thought I've cleaned it all up. New commit now available.

**Review by NoahNsimbe — APPROVED** [review]
Thanks @uman95. The code changes LGTM. Let a suggestion(s) for your consideration.

File: src/gp-model/main.py [hunk 2]
```diff
@@ -234,46 +163,44 @@ def periodic_function(tenant, airqloud, aq_id):
...
+        airqlouds = requests.get(
+            configuration.VIEW_AIRQLOUD_URI, params=params).json()['airqlouds']
     names = [aq['name'] for aq in airqlouds]
     aq_ids = [aq['_id'] for aq in airqlouds]
-    # Esxcluding Uganda
-    del names[-1:]
-    del aq_ids[-1:]
-    return names, aq_ids
+    # Esxcluding Country level AirQloud
+    exclude = ['kenya', 'uganda', 'cameroon', 'senegal', 'gulu']
```
[the remark this answers is not in the document]
Comment (NoahNsimbe) [reply src/gp-model/main.py:198]: Screenshot 2023-03-27 at 22 13 42

[the remark this answers is not in the document]
Comment (uman95) [reply src/gp-model/main.py:198]: Thanks @NoahNsimbe will address it.

File: src/gp-model/config.py [hunk 3]
```diff
@@ -47,6 +47,7 @@ class TestingConfig(Config):
     VIEW_AIRQLOUD_URI = os.getenv('VIEW_AIRQLOUD_URI_STAGE')
     LIST_DEVICES_URI = os.getenv('LIST_DEVICES_URI_STAGE')
     EVENTS_URI = os.getenv('EVENTS_URI_STAGE')
+    API_TOKEN = os.getenv('STAGE_TOKEN')
```
Comment (123MwanjeMike) [inline src/gp-model/config.py:50]: Could we have this API_TOKEN environment independent?

File: src/gp-model/main.py [hunk 4]
```diff
@@ -55,92 +22,54 @@ def get_airqloud_polygon(tenant, airqloud):
     params = {'tenant': tenant,
               'name': airqloud
               }
-    coords = requests.get(configuration.VIEW_AIRQLOUD_URI, params=params).json()[
-        'airqlouds'][0]['location']['coordinates']
+    if environment == 'staging':
+        headers = {'Authorization': configuration.API_TOKEN}
```
Comment (123MwanjeMike) [inline src/gp-model/main.py:26]: We also have a new authentication method that use Access token passed as a query parameter which I find to be a cleaner approach.

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
Great work @uman95
I just a few comments on this.
They may not be a big concern but will leave them here so we don't miss them.
Thanks

**Review by sserurich — COMMENTED** [review]
@uman95 , running PR locally with staging environment fails with this message in the image, might need to catch error incase response from the request doesn't have 'airqloud'
image

**Comment by uman95** [comment]
> @uman95 , running PR locally with staging environment fails with this message in the image, might need to catch error incase response from the request doesn't have 'airqloud' image

Hi @sserurich you need an API token now to test out the code in a staging environment. Also check if you are using **version 2 (v2)** in all your APIs because we are moving to version 2 now.

**Comment by sserurich** [comment]
> @uman95 , running PR locally with staging environment fails with this message in the image, might need to catch error incase response from the request doesn't have 'airqloud' image

The issue stems from token not working.. but in the future, might need to catch this as well.

**Review by sserurich — APPROVED** [review]
LGTM, now

File: src/gp-model/main.py (hunk 4 above)
Comment (uman95) [reply src/gp-model/main.py:26]: @123MwanjeMike done.

File: src/gp-model/config.py (hunk 3 above)
Comment (uman95) [reply src/gp-model/config.py:52]: Thanks @123MwanjeMike I have made it independent.

**Review by 123MwanjeMike — APPROVED** [review]
Great work @uman95 
LGTM

File: src/gp-model/main.py (hunk 4 above)
Comment (123MwanjeMike) [reply src/gp-model/main.py:26]: Great, thanks @uman95

**Pull request merged**